`poly_to_angles` fixing #6978 (#6979)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  passed to the `QNode`, instead of assuming the method is always `deferred`.
  [(#6903)](https://github.com/PennyLaneAI/pennylane/pull/6903)

> * The `poly_to_angles` functionality now correctly works with different tensor interfaces. 
>   Also, now it does not modify the input.

Reviewer: ```suggestion
* The `poly_to_angles` function is improved to correctly work with different tensor interfaces ans also
  to not change the input angles array internally.
```

> * The `poly_to_angles` function is improved to correctly work with different tensor interfaces ans also
>   to not change the input angles array internally.

Reviewer 2: ```suggestion
* The `poly_to_angles` function has been improved to correctly work with different interfaces and
  no longer manipulate the input angles tensor internally.
```

  passed to the `QNode`, instead of assuming the method is always `deferred`.
  [(#6903)](https://github.com/PennyLaneAI/pennylane/pull/6903)

* The `poly_to_angles` function has been improved to correctly work with different interfaces and
  no longer manipulate the input angles tensor internally.
  [(#6979)](https://github.com/PennyLaneAI/pennylane/pull/6979)

* The `QROM` template is upgraded to decompose more efficiently when `work_wires` are not used.
  [#6967)](https://github.com/PennyLaneAI/pennylane/pull/6967)

> 

Author: ```suggestion
```
